numpy/f2py/common_rules.py

```diff
@@ -32,9 +32,7 @@ def findcommonblocks(block, top=1):
     ret = []
     if hascommon(block):
         for n in block['common'].keys():
-            vars = {}
-            for v in block['common'][n]:
-                vars[v] = block['vars'][v]
+            vars = {v: block['vars'][v] for v in block['common'][n]}
```

Comment:
While you're here - changing the outer loop to iterate over `block['common'].items()` would make this loop easier to read